### PR TITLE
[VL] Fix shuffle ignores uncompressed compression configuration

### PR DESCRIPTION
**Pull request by marin-ma**

### CODE REVIEW

**Comment by marin-ma** [comment]
cc: @weiting-chen

File: cpp/core/jni/JniWrapper.cc [hunk 1]
```diff
@@ -785,8 +785,8 @@ JNIEXPORT jlong JNICALL Java_io_glutenproject_vectorized_ShuffleWriterJniWrapper
     shuffleWriterOptions.buffer_size = bufferSize;
   }
 
+  shuffleWriterOptions.compression_type = getCompressionType(env, codecJstr);
   if (codecJstr != NULL) {
-    shuffleWriterOptions.compression_type = getCompressionType(env, codecJstr);
     shuffleWriterOptions.codec_backend = getCodecBackend(env, codecBackendJstr);
     shuffleWriterOptions.compression_mode = getCompressionMode(env, compressionModeJstr);
   }
```
Comment (zhztheplayer) [inline cpp/core/jni/JniWrapper.cc:792]: Shuffle reader has similar code

https://github.com/oap-project/gluten/blob/797770da8da369a55266379fedbca5454680979b/cpp/core/jni/JniWrapper.cc#L994-L997

Should we apply the change to that part too?

Comment (marin-ma) [reply cpp/core/jni/JniWrapper.cc:792]: Thanks for catching this!

**Comment by marin-ma** [comment]
@zhztheplayer Could you help to review again? Thanks!

**Review by zhztheplayer — APPROVED** [review]
Thanks!

**Pull request merged**